clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -3938,25 +3938,15 @@ private Call getAlterConsumerGroupOffsetsCall(ConsumerGroupOperationContext<Map<
             void handleResponse(AbstractResponse abstractResponse) {
                 final OffsetCommitResponse response = (OffsetCommitResponse) abstractResponse;
 
-                // If coordinator changed since we fetched it, retry
-                if (ConsumerGroupOperationContext.hasCoordinatorMoved(response)) {
+                // 1) If coordinator changed since we fetched it, retry
+                // 2) If there is a coordinator error, retry
+                if (ConsumerGroupOperationContext.hasCoordinatorMoved(response) ||
+                    ConsumerGroupOperationContext.shouldRefreshCoordinator(response)) {
```

Comment:
While the code is cleaner, the error counts map (`errorCounts`) is re-computed 3 times now. Once in `hasCoordinatorMoved` and twice in `shouldRefreshCoordinator`. It would be better if we could get it once and reuse it.

Reply:
You are right!